typescript/packages/plugins/coinmarketcap/src/parameters.ts

```diff
@@ -0,0 +1,60 @@
+import { createToolParameters } from "@goat-sdk/core";
+import { z } from "zod";
+
+export class CryptocurrencyListingsParameters extends createToolParameters(
+    z.object({
+        start: z.number().optional().describe("Starting position of results"),
+        limit: z.number().optional().describe("Number of results to return"),
+        sort: z.string().optional().describe("What field to sort the list by"),
+        sort_dir: z.enum(["asc", "desc"]).optional().describe("Direction to sort the list"),
+        cryptocurrency_type: z
+            .enum(["all", "coins", "tokens"])
+            .optional()
+            .describe("Type of cryptocurrency to include"),
+        tag: z.string().optional().describe("Tag to filter by"),
+        aux: z.string().optional().describe("Comma-separated list of auxiliary fields to return"),
+        convert: z.string().optional().describe("Currency to convert prices to"),
+    }),
+) {}
+
+export class CryptocurrencyQuotesLatestParameters extends createToolParameters(
+    z.object({
+        id: z.string().optional().describe("One or more comma-separated cryptocurrency IDs"),
+        symbol: z.string().optional().describe("One or more comma-separated cryptocurrency symbols"),
+        convert: z.string().optional().describe("Currency to convert prices to"),
+        aux: z.string().optional().describe("Comma-separated list of auxiliary fields to return"),
+    }),
+) {}
+
+export class ExchangeListingsParameters extends createToolParameters(
+    z.object({
+        start: z.number().optional().describe("Starting position of results"),
+        limit: z.number().optional().describe("Number of results to return"),
+        sort: z.string().optional().describe("What field to sort the list by"),
+        sort_dir: z.enum(["asc", "desc"]).optional().describe("Direction to sort the list"),
+        market_type: z.enum(["all", "spot", "derivatives"]).optional().describe("Type of exchange market"),
+        aux: z.string().optional().describe("Comma-separated list of auxiliary fields to return"),
```

Comment:
Make this an array. Do the comma separation in the function.

Add the possible values:

num_market_pairs,traffic_score,rank,exchange_score,effective_liquidity_24h,date_launched,fiats